pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -564,114 +590,100 @@ func (service *SecurityPolicyService) buildRuleAppliedGroupByRule(obj *v1alpha1.
 	return &ruleAppliedGroup, ruleAppliedGroupPath, nil
 }
 
-func (service *SecurityPolicyService) buildRuleSrcGroup(obj *v1alpha1.SecurityPolicy, rule *v1alpha1.SecurityPolicyRule, idx int) (*model.Group, string, error) {
-	var ruleSrcGroupName string
-	sources := rule.Sources
-	ruleSrcGroupID := fmt.Sprintf("sp_%s_%d_src", obj.UID, idx)
-	if len(rule.Name) > 0 {
-		ruleSrcGroupName = fmt.Sprintf("%s-src", rule.Name)
+func (service *SecurityPolicyService) buildRulePeerGroupID(obj *v1alpha1.SecurityPolicy, ruleIdx int, isSource bool) string {
+	if isSource == true {
+		return fmt.Sprintf("sp_%s_%d_src", obj.UID, ruleIdx)
 	} else {
-		ruleSrcGroupName = fmt.Sprintf("%s-%d-src", obj.ObjectMeta.Name, idx)
-	}
-	ruleSrcGroupPath := fmt.Sprintf("/infra/domains/%s/groups/%s", getDomain(service), ruleSrcGroupID)
-	peerTags := service.BuildPeerTags(obj, &sources, idx)
-	ruleSrcGroup := model.Group{
-		Id:          &ruleSrcGroupID,
-		DisplayName: &ruleSrcGroupName,
-		Tags:        peerTags,
+		return fmt.Sprintf("sp_%s_%d_dst", obj.UID, ruleIdx)
 	}
+}
 
-	ruleSrcGroupCount, ruleSrcGroupTotalExprCount := 0, 0
-	criteriaCount, totalExprCount := 0, 0
-	var err error = nil
-	errorMsg := ""
-	for i, peer := range sources {
-		criteriaCount, totalExprCount, err = service.updatePeerExpressions(
-			obj,
-			&peer,
-			&ruleSrcGroup,
-			i,
-		)
-		if err == nil {
-			ruleSrcGroupCount += criteriaCount
-			ruleSrcGroupTotalExprCount += totalExprCount
+func (service *SecurityPolicyService) buildRulePeerGroupPath(obj *v1alpha1.SecurityPolicy, groupID string) (string, error) {
```

Comment:
Unused parameter 'obj *v1alpha1.SecurityPolicy'

Reply:
Thanks, after refactoring, we removed parameter: obj *v1alpha1.SecurityPolicy